.github/workflows/packages-e2e-tests.yaml

```diff
@@ -37,19 +41,71 @@ jobs:
         id: tag
         run: echo "tag=$(make version)" >> $GITHUB_OUTPUT
 
-      # Logging into Docker Hub makes the job less flaky, but it relies on repo secrets so
-      # does not work for external contributors. Let's compromise and do this for internal
-      # PRs at least (for now). We can always improve the situation later.
-      - name: Login to Docker Hub
-        if: ${{ github.event.pull_request.head.repo.full_name == 'cilium/tetragon' }}
-        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
-        with:
-          username: ${{ secrets.DOCKER_HUB_USERNAME_CI }}
-          password: ${{ secrets.DOCKER_HUB_PASSWORD_CI }}
-
       - name: Generate Tetragon Tarball
         id: tetragon-tarball
-        run: make tarball
+        run: |
+          make tarball
+          mkdir ${{ matrix.upload_path }}
+          mv ./build/${{ matrix.arch }}/linux-tarball/tetragon-${{ steps.tag.outputs.tag }}-${{ matrix.arch }}.tar.gz ./${{ matrix.upload_path }}
+        if: ${{ matrix.cross_compile == 'no' }}
+
+      - name: Generate Cross Compiled Tetragon EE Tarball
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu qemu-user-static binfmt-support
+          sudo update-binfmts --display
+          make TARGET_ARCH=${{ matrix.arch }} tarball
+          mkdir ${{ matrix.upload_path }}
+          mv ./build/${{ matrix.arch }}/linux-tarball/tetragon-${{ steps.tag.outputs.tag }}-${{ matrix.arch }}.tar.gz ./${{ matrix.upload_path }}
+        if: ${{ matrix.cross_compile == 'yes' }}
```

Comment:
To avoid this another var you could also use to compare:
> `RUNNER_ARCH` The architecture of the runner executing the job. Possible values are X86, X64, ARM, or ARM64.

not sure it's easier or better tbh